packages/react-devtools-shared/src/__tests__/profilingCache-test.js

```diff
@@ -752,4 +752,97 @@ describe('ProfilingCache', () => {
     utils.act(() => store.profilerStore.stopProfiling());
     expect(container.textContent).toBe('About');
   });
+
+  it('components that were deleted and added to updaters during the layout phase should not crash', () => {
+    let setBarUnmounted;
+    function Bar() {
+      const [, setState] = React.useState(false);
+
+      React.useLayoutEffect(() => {
+        return () => setState(true);
+      });
+
+      return null;
+    }
+
+    function App() {
+      const [barUnmounted, _setBarUnmounted] = React.useState(false);
+      setBarUnmounted = _setBarUnmounted;
+      return <>{!barUnmounted && <Bar />}</>;
+    }
+
+    const root = ReactDOM.createRoot(document.createElement('div'));
+    utils.act(() => root.render(<App />));
+    utils.act(() => store.profilerStore.startProfiling());
+    utils.act(() => setBarUnmounted(true));
+    utils.act(() => store.profilerStore.stopProfiling());
+
+    const updaters = store.profilerStore.getCommitData(store.roots[0], 0)
+      .updaters;
+    expect(updaters.length).toEqual(1);
+    expect(updaters[0].displayName).toEqual('App');
+  });
+
+  it('components in a deleted subtree and added to updaters during the layout phase should not crash', () => {
+    function Foo() {
+      return <Bar />;
+    }
+
+    let setBarUnmounted;
+    function Bar() {
```

Comment:
nit: You could pick slightly more meaningful names here ("Child" and "Grandchild" instead of "Foo" and "Bar") but that's a nit – no big deal.